_posts/2018-04-25-productivity-with-scripts-screencapture.md

```diff
@@ -0,0 +1,153 @@
+
+# Investigation Time
+
+After learning of the `screencapture` utility, I began to explore its docs and try it out. I first identified what I wanted to accomplish with my script, and then looked for corresponding flags or variations on the command that would support the behavior I wanted.
+
+I knew I wanted to specify a name and extension for my file on the fly. The following is from `man screencapture`.
+```
+SYNOPSIS
+     screencapture [-SWCTMPcimswxto] file
+```
+I learned `<file>` is the path to destination of where the screenshot will be saved. I also knew that I wanted to select only a portion of the screen with my mouse when taking the screenshot. After exploring the documentation for this utility more I found the following flag.
+```
+-s      Only allow mouse selection mode.
+```
+So I came to the conclusion that I would be invoking the `screencapture` utility from my script as follows.
+```
+screencapture -s <path_to_file>
+```
+I felt comfortable enough with the utility to proceed after identifying how it would be used. All that was left was a matter of writing the bash script and making the file and extension names dynamic.
+
+I also decided that it would be very helpful to have the `<path_to_file>` copied to my clipboard. This would make the file easy to open in preview on the command line after the script ran with the command: `open <path_to_file>`. Having the path to the clipboard would also make it easier to locate in `Finder` for Mac (You can enter absolute path in `Finder` with `command` + `shift` + `G`).
+Luckily, Mac has the `pbcopy` command that allows you to paste to your system `Clipboard` via the command line. You can read more about `pbcopy` by typing `man pbcopy` in your terminal.
+
+# Writing The Script
+
+I will not go a ton into bash scripting in this tutorial. There are plenty of other great tutorials and resources for that online. However, I will provide a brief description of my thought process and detailed comments within the attached snippet for the script.
+
+After looking into the `screencapture` utility, I learned that it requires one argument (`<path_to_file>`). This means that my script would at minimum require one argument as well. I didn't want to type out an absolute path or always invoke the script from my screenshot directory, so I decided I would only pass the file's base name to my script, and keep the path to my desired directory in a variable. I decided to concatenate the base file name argument with my desired directory within my script and pass this resulting absolute path to the `screencapture` command.
+
+I decided that I may also want to specify the extension of the file. I'm usually fine with `png` as an extension. To limit keystrokes, I decided that this would be the default extension for my screenshots. I also decided to make the extension a second optional argument to my script.
```

Comment:
Lots of "I learned" and "I decided" in here. You can eliminate those, and just offer them up as facts, which they are.